mm2src/coins/utxo/utxo_common.rs

```diff
@@ -2250,6 +2252,29 @@ where
     }
 }
 
+pub fn save_transaction_history<T>(
```

Comment:
How about instead of a new method I just extend the `MmCoin::save_history_to_file` with:

```
let mut to_write: Vec<TransactionDetails> = history_map.iter().map(|(_, value)| value.clone()).collect();
// the transactions with block_height == 0 are the most recent so we need to separately handle them while sorting
to_write.sort_unstable_by(|a, b| {
    if a.block_height == 0 {
        Ordering::Less
    } else if b.block_height == 0 {
        Ordering::Greater
    } else {
        b.block_height.cmp(&a.block_height)
    }
});
```

It seems like everytime before we write to the history we do the same sorting except for QRC20. There we also sort according to internal ids in case the block heights are equal:
```

to_write.sort_unstable_by(|a, b| {
    match sort_newest_to_oldest(a.block_height, b.block_height) {
        // do not reverse `transfer` events in one transaction
        Ordering::Equal => a.internal_id.cmp(&b.internal_id),
        ord => ord,
    }
});
```

I think I could use the `MmCoin::save_history_to_file` and at the beginning of the method sort the transactions as in the QRC20 for all coins. What do you think?

Reply:
I thought about it, but if I'm not mistaken, `EthCoin` uses `MmCoin::save_history_to_file` as well, but it's implemented little different from UTXO/QRC20 coins.
If it's possible, let's do it!
>  I just extend the MmCoin::save_history_to_file

Reply:
I didn't see any difference of `EthCoin` in the way it uses the `MmCoin::save_history_to_file`, so I made the sorting part of the method for all coins.